docs/advanced/accesstransformers.md

```diff
@@ -0,0 +1,111 @@
+--------
+
+All text after a `#` until the end of the line will be treated as a comment and will not be parsed.
+
+Access Modifiers
+----------------
+
+Access modifiers specify which member visibility the given target will be transformed to. In decreasing order of visibility:
+
+  * `public` - visible to all classes inside and outside its package
+  * `protected` - visible only to classes inside the package and subclasses
+  * `default` - visible only to classes inside the package
+  * `private` - visible only to inside the class
+
+A special modifier `+f` and `-f` can be appended to the aforementioned modifiers to either add or remove respectively the `final` modifier, which prevents subclassing, method overriding, or field modification when applied.
+
+!!! warning
+    Access transformers can **_never_** make a member less visible; they can only widen visibility. The JVM will throw an `IllegalAccessError` if any compiled code references a method or field which cannot be accessed due to restrictive transformations.
+    
+    Directives only modify the method they directly reference; any overriding methods will _not_ be access-transformed. Therefore, **never** transform any non-`public` or non-`static` method, because any overriding methods might still use the restrictive visibility, which will result in the JVM throwing a `VerifyError`. 
+
+Targets and Directives
+------------------
+
+!!! Information
+    When using Access Transformers on Minecraft classes, the SRG name must be used for fields and methods.
+
+### Classes
+To target classes:
+```
+<access modifier> <fully qualified class name>
+```
+Inner classes are denoted by combining the fully qualified name of the outer class and the name of the inner class with a `$` as separator.
+
+### Fields
+To target fields:
+```
+<access modifier> <fully qualified class name> <field name>
+```
+
+### Methods
+Targeting methods require a special syntax to denote the method parameters and return type:
+```
+<access modifier> <fully qualified class name> <method name>(<parameter types>)<return type>
+```
+
+#### Specifying Types
+
+Also called "descriptors": see the [Java Virtual Machine Specification, SE 8, sections 4.3.2 and 4.3.3][jvmdescriptors] for more technical details.
+
+If the return type is void, or the method has no parameters, then it does not need to be specified.
+
+  * `B` - `byte`, a signed byte
+  * `C` - `char`, a Unicode character code point in UTF-16
+  * `D` - 'double', a double-precision floating-point value
+  * `F` - `float`, a single-precision floating-point value
+  * `I` - `integer`, a 32-bit integer
+  * `J` - `long`, a 64-bit integer
+  * `S` - `short`, a signed short
+  * `Z` - `boolean`, a `true` or `false` value
+  * `[` - references one dimension of an array
+    * Example: `[[S` refers to `short[][]`
+  * `L<class name>;` - references a reference type
+    * Example: `Ljava/lang/String;` refers to `java.lang.String` reference type _(note the use of parentheses)_
```

Comment:
You mean slashes, not parentheses, right?

Reply:
Yep, will correct that mistake.